### PR TITLE
Fix CacheManagerResource reflection

### PR DESCRIPTION
**Pull request by ryanemerson**

### CODE REVIEW

**Comment by ryanemerson** [comment]
Follow up PR on Infinispan master https://github.com/infinispan/infinispan/pull/8473

Once ^ has been merged, I will remove the test from this repository and simply reference `RestCacheManagerResource` from `NativeClusteredIT`.

File: server/deployment/src/main/java/org/infinispan/quarkus/server/deployment/InfinispanServerProcessor.java [hunk 1]
```diff
@@ -173,6 +173,12 @@ void addReflectionAndResources(BuildProducer<ReflectiveClassBuildItem> reflectio
       registerClass(reflectionClass, MemoryType.class, true, false, true);
       registerClass(reflectionClass, MemoryUsage.class, true, false, true);
 
+      reflectionClass.produce(new ReflectiveClassBuildItem(true, false, "org.infinispan.health.impl.CacheHealthImpl"));
```
Comment (wburns) [inline server/deployment/src/main/java/org/infinispan/quarkus/server/deployment/InfinispanServerProcessor.java:176]: I prefer referencing the class when possible so we are aware of changes in package/name etc.

Comment (ryanemerson) [reply server/deployment/src/main/java/org/infinispan/quarkus/server/deployment/InfinispanServerProcessor.java:176]: It's a package-private class.

Comment (wburns) [reply server/deployment/src/main/java/org/infinispan/quarkus/server/deployment/InfinispanServerProcessor.java:176]: K cool, just wanted to make sure.

**Pull request merged**